src/Utilities/Compiler/Options/EditorConfigOptionNames.cs

```diff
@@ -204,5 +204,17 @@ internal static partial class EditorConfigOptionNames
         /// CA1826 (Do not use Enumerable methods on indexable collections. Instead use the collection directly).
         /// </summary>
         public const string ExcludeOrDefaultMethods = "exclude_ordefault_methods";
+
+        /// <summary>
+        /// String option to configure names of method symbols (separated by '|') that marks all of the parameters with IEnumerable type
+        /// would be not enumerated.
+        /// </summary>
+        public const string NoEnumerationMethods = "no_enumeration_methods";
+
+        /// <summary>
+        /// String option to configure names of method symbols (separated by '|') that marks all of the parameters with IEnumerable type
+        /// would be not enumerated, and return a new IEnumerable type.
```

Comment:
I might misunderstood, but it seems to me `that marks all of the parameters with IEnumerable type would be not enumerated` part is not the requirement, seems it is only about `a method that return a new IEnumerable type.`, no?